### ISSUE TITLE
Read data from s3

### ISSUE THREAD

**Issue by JulienPeloton** [astrolabsoftware/spark-fits#65]
I have been trying to read FITS files in a Ceph cluster from s3. It throws the following error:

```
Py4JJavaError: An error occurred while calling o63.load.
: java.io.EOFException: Attempted to seek or read past the end of the file 741911040
	at org.apache.hadoop.fs.s3a.S3AInputStream.reopen(S3AInputStream.java:81)
	at org.apache.hadoop.fs.s3a.S3AInputStream.seek(S3AInputStream.java:115)
	at org.apache.hadoop.fs.FSDataInputStream.seek(FSDataInputStream.java:62)
	at com.astrolabsoftware.sparkfits.FitsLib$Fits.getNHDU(FitsLib.scala:404)
	at com.astrolabsoftware.sparkfits.FitsLib$Fits.<init>(FitsLib.scala:148)
	at com.astrolabsoftware.sparkfits.FitsRelation$$anonfun$schema$1.apply(FitsSourceRelation.scala:364)
	at com.astrolabsoftware.sparkfits.FitsRelation$$anonfun$schema$1.apply(FitsSourceRelation.scala:360)
	at scala.Option.getOrElse(Option.scala:121)
	at com.astrolabsoftware.sparkfits.FitsRelation.schema(FitsSourceRelation.scala:360)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:431)
	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:239)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:227)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:174)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
```
I'm not sure whether this is just a bug from the code (like index mismatch that was not caught by e.g. hadoop), or one needs to implement new things... First thing to look at: `getNHDU`.

**Comment by jacobic** [comment]
Hey man, hope you are well. 

It looks like you solved this issue is #66 but I get an error that is very similar (although nothing to do with s3). I am really struggling to diagnose it but here is the output:

My only guess is that there is some sort of corruption and that I should recreate the fits files do you have any advice?

```
Job aborted due to stage failure: Task 3040 in stage 0.0 failed 4 times, most recent failure: Lost task 3040.3 in stage 0.0 (TID 3141, 130.183.20.39, executor 12): java.io.EOFException: Cannot seek after EOF
	at org.apache.hadoop.fs.ChecksumFileSystem$FSDataBoundedInputStream.seek(ChecksumFileSystem.java:328)
	at com.astrolabsoftware.sparkfits.FitsLib$Fits$$anonfun$getNHDU$4.apply$mcV$sp(FitsLib.scala:428)
	at com.astrolabsoftware.sparkfits.FitsLib$Fits$$anonfun$getNHDU$4.apply(FitsLib.scala:428)
	at com.astrolabsoftware.sparkfits.FitsLib$Fits$$anonfun$getNHDU$4.apply(FitsLib.scala:428)
	at scala.util.Try.getOrElse(Try.scala:79)
	at com.astrolabsoftware.sparkfits.FitsLib$Fits.getNHDU(FitsLib.scala:428)
	at com.astrolabsoftware.sparkfits.FitsLib$Fits.<init>(FitsLib.scala:148)
	at com.astrolabsoftware.sparkfits.FitsRecordReader.initialize(FitsRecordReader.scala:162)
	at org.apache.spark.rdd.NewHadoopRDD$$anon$1.liftedTree1$1(NewHadoopRDD.scala:199)
	at org.apache.spark.rdd.NewHadoopRDD$$anon$1.<init>(NewHadoopRDD.scala:196)
	at org.apache.spark.rdd.NewHadoopRDD.compute(NewHadoopRDD.scala:151)
	at org.apache.spark.rdd.NewHadoopRDD.compute(NewHadoopRDD.scala:70)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```

Thanks in advance,
Jacob

**Comment by JulienPeloton** [comment]
Hi @jacobic ! Sorry for the delay. I never encountered such error in the past.
Could you describe the input data set? What is the size on disk of it, number of files, and structure (columns and rows size)?
